### ISSUE TITLE
redirect chain is incomplete with firefox [BUG]

### ISSUE THREAD

**Issue by sammyyu** [microsoft/playwright#6848]
**Context:**
- Playwright Version: 1.11.1
- Operating System: Linux
- Node.js version: [e.g. 12.22, 14.6]
- Browser: Firefox
- Extra: [any specific details about your environment]

**Code Snippet**

```
    redirectChainList = []
    if (basePageRequest!=null) {
      for (let r = basePageRequest; r; r = r.redirectedTo()) {
        redirectStatusCode = 301
        let currentURL = r.url()
        let nextURL = null
        if (r.redirectedTo()) {
          nextURL = r.redirectedTo().url()
        }
        redirectChainList.push([r.url(), nextURL, redirectStatusCode]);
  }
```

**Describe the bug**
I'm trying to get the redirect chain for the website: https://www.myanimelist.net/.

I'm saving the original request and following the redirectTo()  using the the code above (ignore the status code for now):
It comes up with an incomplete chain
`
["https://www.myanimelist.net/","http://myanimelist.net/",301],["http://myanimelist.net/",null,301]
`

If I run the same thing in webkit or chrome it comes up with correct chain
`
["https://www.myanimelist.net/","http://myanimelist.net/",301],["http://myanimelist.net/","https://myanimelist.net/",301],["https://myanimelist.net/",null,301]
`

**Comment by pavelfeldman** [comment]
I can reproduce:

```js
const { chromium, firefox, webkit } = require('playwright');

(async () => {
  const browser = await firefox.launch({});
  const page = await browser.newPage();
  const response = await page.goto('https://www.myanimelist.net');
  for (let request = response.request(); request; request = request.redirectedFrom())
    console.log(request.url(), (await request.response()).status());
  await browser.close();
})();
```

**Comment by dgozman** [comment]
This should be fixed by #7419 in v1.13.
